NitroxClient/Unity/Helper/GUISkinUtils.cs

```diff
@@ -0,0 +1,139 @@
+
+        public static GUISkin CreateDerived(GUISkin baseSkin = null, string name = null)
+        {
+            if (baseSkin == null)
+            {
+                GUISkin prevSkin = GUI.skin;
+                GUI.skin = null;
+                baseSkin = GUI.skin;
+                GUI.skin = prevSkin;
+            }
+            
+            GUISkin copy = ScriptableObject.CreateInstance<GUISkin>();
+
+            copy.name = name ?? Guid.NewGuid().ToString();
+            copy.box = new GUIStyle(baseSkin.box);
+            copy.button = new GUIStyle(baseSkin.button);
+            copy.label = new GUIStyle(baseSkin.label);
+            copy.scrollView = new GUIStyle(baseSkin.scrollView);
+            copy.textArea = new GUIStyle(baseSkin.textArea);
+            copy.textField = new GUIStyle(baseSkin.textField);
+            copy.toggle = new GUIStyle(baseSkin.toggle);
+            copy.window = new GUIStyle(baseSkin.window);
+            copy.horizontalScrollbar = new GUIStyle(baseSkin.horizontalScrollbar);
+            copy.horizontalScrollbarLeftButton = new GUIStyle(baseSkin.horizontalScrollbarLeftButton);
+            copy.horizontalScrollbarRightButton = new GUIStyle(baseSkin.horizontalScrollbarRightButton);
+            copy.horizontalScrollbarThumb = new GUIStyle(baseSkin.horizontalScrollbarThumb);
+            copy.horizontalSlider = new GUIStyle(baseSkin.horizontalSlider);
+            copy.horizontalSliderThumb = new GUIStyle(baseSkin.horizontalSliderThumb);
+            copy.verticalScrollbar = new GUIStyle(baseSkin.verticalScrollbar);
+            copy.verticalScrollbarDownButton = new GUIStyle(baseSkin.verticalScrollbarDownButton);
+            copy.verticalScrollbarThumb = new GUIStyle(baseSkin.verticalScrollbarThumb);
+            copy.verticalScrollbarUpButton = new GUIStyle(baseSkin.verticalScrollbarUpButton);
+            copy.verticalSlider = new GUIStyle(baseSkin.verticalSlider);
+            copy.verticalSliderThumb = new GUIStyle(baseSkin.verticalSliderThumb);
+            copy.customStyles = baseSkin.customStyles.Select(s => new GUIStyle(s)).ToArray();
+            copy.hideFlags = baseSkin.hideFlags;
+
+            copy.settings.cursorColor = baseSkin.settings.cursorColor;
+            copy.settings.cursorFlashSpeed = baseSkin.settings.cursorFlashSpeed;
+            copy.settings.doubleClickSelectsWord = baseSkin.settings.doubleClickSelectsWord;
+            copy.settings.tripleClickSelectsLine = baseSkin.settings.tripleClickSelectsLine;
+            copy.settings.selectionColor = baseSkin.settings.selectionColor;
+
+            // TODO: Create identical copy of font.
+            copy.font = baseSkin.font; // Gives a giberish font: copy.font = UnityEngine.Object.Instantiate(baseSkin.font);
+
+            return copy;
+        }
+
+        public static GUISkin RegisterDerived(string name, GUISkin baseSkin = null)
+        {
+            if (name == null)
```

Comment:
We prefere to use the `Validate.NotNull` utility method for null checks (and `Validate.IsTrue/False` a couple lines below).